src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala

```diff
@@ -105,20 +110,122 @@ class AnxAnalyses(format: AnxAnalysisStore.Format) {
   private[this] val reader = new ProtobufReaders(mappers.getReadMapper, schema.Version.V1)
   private[this] val writer = new ProtobufWriters(mappers.getWriteMapper)
 
+  def sortAnnotation(annotation: Annotation): Annotation = {
+    annotation.copy(
+      arguments = annotation.arguments.sortWith(_.hashCode() > _.hashCode())
```

Comment:
My protobuf knowledge isn't super great, but does the `Type` in `Annotation` also need to be sorted? 

Type in annotation: https://github.com/sbt/zinc/blob/develop/internal/zinc-persist-core/src/main/protobuf/schema.proto#L272

Type: https://github.com/sbt/zinc/blob/develop/internal/zinc-persist-core/src/main/protobuf/schema.proto#L210

Reply:
Hmm, I was looking at the decompiled `.class` files to see what attributes to sort - the Type class looks like this: 
```
public abstract class Type implements Serializable {
    protected Type() {
    }

    public boolean equals(Object var1) {
        if (this == var1) {
            return true;
        } else if (!(var1 instanceof Type)) {
            return false;
        } else {
            Type var2 = (Type)var1;
            return true;
        }
    }

    public int hashCode() {
        return 37 * (17 + "xsbti.api.Type".hashCode());
    }

    public String toString() {
        return "Type()";
    }
}
```
which doesn't seem to have anything, really. What am I missing here?

Reply:
Upon further reflection, I've noticed that this is an abstract class and therefore there are a bunch of different implementations of `Type`. I'm going to look into somehow sorting them